src/modules/75-cf/components/FlagArchiving/useRestoreFlagDialog.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { Intent } from '@harness/design-system'
+import { UseConfirmationDialogReturn, useToaster } from '@harness/uicore'
+import { String, useStrings } from 'framework/strings'
+import { useRestoreFeatureFlag, RestoreFeatureFlagQueryParams, Feature } from 'services/cf'
+import useResponseError from '@cf/hooks/useResponseError'
+import { useConfirmAction } from '@common/hooks'
+import css from './useRestoreFlagDialog.module.scss'
+
+export interface RestoreFlagDialogProps {
+  flagData: Feature
+  queryParams: RestoreFeatureFlagQueryParams
+  onRefetch: () => void
+}
+
+const useRestoreFlagDialog = ({
+  flagData,
+  queryParams: restoreQueryParams,
+  onRefetch
+}: RestoreFlagDialogProps): UseConfirmationDialogReturn['openDialog'] => {
+  const { showSuccess, showError } = useToaster()
+  const { getString } = useStrings()
+  const { handleResponseError } = useResponseError()
+
+  const queryParams = {
+    identifier: flagData.identifier,
+    environmentIdentifier: flagData.envProperties?.environment,
+    ...restoreQueryParams
+  }
+
+  const { mutate: restoreFeatureFlag } = useRestoreFeatureFlag({
+    identifier: flagData.identifier,
+    queryParams
+  })
+
+  const handleRestore = async (): Promise<void> => {
+    try {
+      await restoreFeatureFlag()
+      showSuccess(getString('cf.featureFlags.archiving.restoreSuccess'))
+      onRefetch()
+    } catch (e) {
+      handleResponseError(e)
+      showError(e.message)
+    }
+  }
+
+  return useConfirmAction({
+    title: getString('cf.featureFlags.archiving.restoreFlag'),
+    intent: Intent.DANGER,
+    confirmText: getString('cf.featureFlags.archiving.restore'),
+    message: (
+      <span className={css.restoreMessage}>
+        <String stringID="cf.featureFlags.archiving.restoreDescription" vars={{ flagName: flagData.name }} />
```

Comment:
Out of curiosity, why use the `<String />` component instead of just `getString`?
```suggestion
        {getString('cf.featureFlags.archiving.restoreDescription', { flagName: flagData.name })}
```

Reply:
just habit since it what I used whenever I needed a string with a var